pyrealm/core/solar.py

```diff
@@ -1,26 +1,584 @@
+        k_A         : calculation constant for Rnl
+        tc          : temperature of observations, °C
+
+    Returns:
+        rnl         : net longwave radiation, W/m^2
+    """
+
+    rnl = (k_b + (1.0 - k_b) * sf) * (k_A - tc)
+
+    return rnl
+
+
+def calc_rw(k_alb_sw: float, tau: NDArray, k_Gsc: float, dr: NDArray) -> NDArray:
+    r"""Calculates variable substitute rw, W/m^2.
+
+    .. math::
+
+        rw = (1.0 - k\_alb\_sw) \cdot \tau \cdot k\_Gsc \cdot dr
+
+    Args:
+        k_alb_sw    : shortwave albedo
+        tau         : bulk transmissivity, unitless
+        k_Gsc       : solar constant, W/m^2
+        dr          : distance ration, unitless
+
+    Returns:
+        rw          : intermediate variable, W/m^2
+    """
+
+    rw = (1.0 - k_alb_sw) * tau * k_Gsc * dr
+
+    return rw
+
+
+def calc_net_rad_crossover_hour_angle(
+    rnl: NDArray,
+    k_pir: float,
+    k_alb_sw: float,
+    tau: NDArray,
+    k_Gsc: float,
+    dr: NDArray,
+    delta: NDArray,
+    latitude: NDArray,
+) -> NDArray:
+    r"""Calculates the net radiation crossover hour angle, degrees.
+
+    This function calculates the net radiation crossover hour angle in degrees.
+
+    .. math::
+
+        hn = \frac{\arccos(\text{clip}((rnl - rw \cdot ru) / (rw \cdot rv), -1.0, 1.0))}
+        {k\_pir}
+
+    Args:
+        rnl         : net longwave radiation, W/m^2
+        k_pir       : conversion factor from radians to degrees
+        k_alb_sw    : shortwave albedo
+        tau         : bulk transmissivity, unitless
+        k_Gsc       : solar constant, W/m^2
+        dr          : distance ration, unitless
+        delta       : solar declination delta
+        latitude    : site latitude(s)
+
+
+    Returns:
+        _calc_net_rad_crossover_hour_angle
+    """
+
+    ru, rv = calc_lat_delta_intermediates(delta, latitude)
+    rw = calc_rw(k_alb_sw, tau, k_Gsc, dr)
+
+    return _calc_net_rad_crossover_hour_angle(rnl, rw, ru, rv, k_pir)
+
+
+def _calc_net_rad_crossover_hour_angle(
+    rnl: NDArray, rw: NDArray, ru: NDArray, rv: NDArray, k_pir: float
+) -> NDArray:
+    """Calculates the net radiation crossover hour angle, degrees.
+
+    This function calculates the net radiation crossover hour angle in degrees.
+
+    Args:
+        rnl         : net longwave radiation, W/m^2
+        rw          : intermediate variable, W/m^2
+        ru          : intermediate variable, W/m^2
+        rv          : intermediate variable, W/m^2
+        k_pir       : conversion factor from radians to degrees
+
+    Returns:
+        hn          : net radiation crossover hour angle, degrees
+    """
+
+    hn = np.arccos(np.clip((rnl - rw * ru) / (rw * rv), -1.0, 1.0)) / k_pir
+
+    return hn
+
+
+def calc_daytime_net_radiation(
+    hn: NDArray,
+    k_pir: float,
+    rnl: NDArray,
+    delta: NDArray,
+    latitude: NDArray,
+    k_alb_sw: float,
+    tau: NDArray,
+    k_Gsc: float,
+    dr: NDArray,
```

Comment:
Maybe order these to put the constants at the end? This might be another helper function that could take `const`?